ciao-controller/instance.go

```diff
@@ -153,12 +164,32 @@ func (c *config) GetResources() map[string]int {
 	return resources
 }
 
-func getStorage(c *controller, wl *types.Workload, tenant string, instanceID string) (payloads.StorageResources, error) {
-	s := wl.Storage
+func addBlockDevice(c *controller, tenant string, instanceID string, device storage.BlockDevice, s types.StorageResource) (payloads.StorageResource, error) {
```

Comment:
Arrgh..the confusion from having a types.StorageResource and a payloads.Storage resource that are very similar.